dev-tools/omdb/tests/test_all_output.rs

```diff
@@ -504,3 +509,19 @@ async fn do_run_extra<F>(
 
     write!(output, "=============================================\n").unwrap();
 }
+
+// OMDB environment variables can affect even the help output provided by clap.
+// See clap-rs/clap#5673 for an example.  This can cause spurious test failures.
+// Clear all the OMDB-related variables from the current process so that all
+// child processes don't have them.
+fn clear_omdb_env() {
+    // Rust documents that this is not safe to do in a multi-threaded process
+    // outside of Windows.  We cannot be sure that we are single-threaded,
+    // and we won't be if we're run with `cargo test`.  But we will be
+    // single-threaded in `cargo nextest` (the officially supported runner).
```

Comment:
So "single-threaded" isn't quite accurate here... libtest will spin up a thread for the test on top of the main thread. But this is completely fine here because the main thread will just be blocked waiting for the test thread.

The important fact here is just that with nextest, we know that nothing else is altering the environment at the same time. So I'd maybe rephrase this in terms of "process-per-test".

Reply:
And there's also separate safety and correctness issues here.

* Two threads using the Rust `std::env::remove_var` to alter the environment is safe but may not be correct.
* One thread using `std::env::remove_var` is both safe and probably correct.
* One thread using `std::env::remove_var` and another thread using libc's `unsetenv` is unsafe on many Unix platforms (though it sounds like not illumos).

Reply:
I edited this comment (and the one before it, since it was implying that the clap issue was the only one here, which isn't true)